src/js/angular/import/directives/import-resource-tree.directive.js

```diff
@@ -214,6 +232,23 @@ function importResourceTreeDirective($timeout, ImportContextService) {
                 const hasSelectedDisplayedImportResource = $scope.displayResources.some((resource) => resource.selected);
                 $scope.areAllDisplayedImportResourcesSelected = hasSelectedDisplayedImportResource && !hasUnselectedDisplayedImportResource;
                 $scope.areAllDisplayedImportResourcesPartialSelected = hasSelectedDisplayedImportResource && hasUnselectedDisplayedImportResource;
+
+                if ($scope.areAllDisplayedImportResourcesSelected) {
+                    $scope.selectedByStatus = STATUS_OPTIONS.ALL;
+                } else if (!$scope.areAllDisplayedImportResourcesSelected && !$scope.areAllDisplayedImportResourcesPartialSelected) {
+                    $scope.selectedByStatus = STATUS_OPTIONS.NONE;
+                }
+            };
+
+            const toggleCheckboxes = (checkboxElement, shouldSelect) => {
+                $scope.displayResources.forEach((resource) => resource.selected = shouldSelect);
+                $scope.selectedByStatus = shouldSelect ? STATUS_OPTIONS.ALL : STATUS_OPTIONS.NONE;
+                $scope.areAllDisplayedImportResourcesSelected = shouldSelect;
+                $scope.areAllDisplayedImportResourcesPartialSelected = false;
+
+                if (checkboxElement) {
+                    checkboxElement.checked = shouldSelect;
```

Comment:
I don't like this. Why are you working with the html element here when it's state is bound to some property in the directive?! See `areAllDisplayedImportResourcesSelected`!